drivers/sensor/adxl375/adxl375.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2023 Aaron Chan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_adxl375
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <string.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/sys/__assert.h>
+#include <stdlib.h>
+#include <zephyr/logging/log.h>
+
+#include "adxl375.h"
+
+LOG_MODULE_REGISTER(ADXL375, CONFIG_SENSOR_LOG_LEVEL);
+
+static int adxl375_check_id(const struct device *dev)
+{
+	struct adxl375_data *data = dev->data;
+	uint8_t device_id = 0;
+
+	int ret = data->hw_tf->read_reg(dev, ADXL375_DEVID, &device_id);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	if (ADXL375_DEVID_VAL != device_id) {
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int adxl375_set_odr_and_lp(const struct device *dev, uint32_t data_rate,
```

Comment:
I cant tell what some of these methods are for by their name alone. Add comments with a brief description of what they do.

Reply:
Oh I didn't see the params, my bad.

Reply:
Btw ODR is common terminology for Output Data Rate. I guess lp might be a little more harder to determine though, although its a common abbreviation for low power